platform/macos/src/MGLMapView.mm

```diff
@@ -1695,6 +1696,21 @@ - (void)setRotateEnabled:(BOOL)rotateEnabled {
     _compass.hidden = !rotateEnabled;
 }
 
+- (void)adjustZoomLevelForKey:(NSString *)key {
+    static NSDictionary<NSString *, void(^)(void)> *zoomOpDict = nil;
+
+    if (zoomOpDict == nil) {
+        void (^zoomOut)(void) = ^ { [self setZoomLevel:(self.zoomLevel - 1) animated:YES]; };
+        void (^zoomIn)(void) = ^ { [self setZoomLevel:(self.zoomLevel + 1) animated:YES]; };
+        zoomOpDict = @{@"-": zoomOut, @"+": zoomIn, @"=": zoomIn};
```

Comment:
Perhaps you could simplify this code with a series of if/else statements that sets a local `newZoomLevel` variable, followed by a single call to `-setZoomLevel:`. If calling `-[NSString isEqualToString:]` in multiple places is undesirable, then you could also switch on the string’s first character.